how-to/networking/dnssec-troubleshooting.md

```diff
@@ -0,0 +1,260 @@
+    ;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
+    ;; WHEN: Fri Oct 25 17:36:25 UTC 2024
+    ;; MSG SIZE  rcvd: 74
+
+This time:
+
+ * There is no `ad` flag set in the answer.
+ * The status of the query is a generic SERVFAIL, and zero answers were provided.
+
+We can tell the Validating Resolver (the service running on the `@127.0.0.1` address) that we don't want it to perform DNSSEC validation. We do that by setting the `+cd` (check disabled) flag. Then things change in our answer:
+
+    $ dig @127.0.0.1 -t A +dnssec +cd +multiline dnssec-failed.org
+
+    ; <<>> DiG 9.18.28-0ubuntu0.24.04.1-Ubuntu <<>> @127.0.0.1 -t A +dnssec +cd +multiline dnssec-failed.org
+    ; (1 server found)
+    ;; global options: +cmd
+    ;; Got answer:
+    ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 7269
+    ;; flags: qr rd ra cd; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 1
+
+    ;; OPT PSEUDOSECTION:
+    ; EDNS: version: 0, flags: do; udp: 1232
+    ; COOKIE: dd66930044348f2501000000671bd808f6852a18a0089b3f (good)
+    ;; QUESTION SECTION:
+    ;dnssec-failed.org.     IN A
+
+    ;; ANSWER SECTION:
+    dnssec-failed.org.      297 IN A 96.99.227.255
+    dnssec-failed.org.      297 IN RRSIG A 5 2 300 (
+                                    20241111145122 20241025144622 44973 dnssec-failed.org.
+                                    fa53BQ7HPpKFIPKyn3Md4bVLawQLeatny47hTq1QouG8
+                                    DwyVqmsfs3d5kUTFO5FHdCy4U7o97ODYXiVuilEZS/aZ
+                                    n6odin2SCm0so4TnIuKBgZFW41zpI6oIRmIVPv6HLerI
+                                    uUxovyMEtaGyd5maNgxGldqLzgWkl18TWALYlrk= )
+
+    ;; Query time: 0 msec
+    ;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
+    ;; WHEN: Fri Oct 25 17:40:24 UTC 2024
+    ;; MSG SIZE  rcvd: 267
+
+Looks like we have some sort of answer, but:
+
+ * There is no `ad` flag in the answer, so this data was not authenticated.
+ * The status is NOERROR, and we got two answers.
+ * Note there is a `cd` flag in the answer, meaning "check disabled". No attempt at validating the answer was done by the resolver, as requested.
+
+In none of these cases, though, did `dig` perform DNSSEC validation: it just presented the results provided by the Validating Resolver, which in some cases was no validation at all (via the `+cd` flag). To perform the validation ourselves, we have to use a different tool.
+
+## Digging a bit deeper: `delv`
+
+The `delv` tool is very similar to `dig`, and can perform the same DNS queries, but with a crucial difference: it can also validate DNSSEC. This is very useful in troubleshooting, because rather than returning a generic SERVFAIL error when something goes wrong with DNSSEC validation, it can tell us what was wrong in more detail.
+
+But to bring the responsibility of doing DNSSEC validation to the tool itself, we use the `+cd` flag in our queries, to tell the resolver to not attempt that validation. Otherwise we will just get back a generic SERVFAIL error:
```

Comment:
```suggestion
But to bring the responsibility of doing DNSSEC validation to the tool itself, we use the `+cd` flag in our queries, to tell the resolver to not attempt that validation. Otherwise we will just get back a generic `SERVFAIL` error:
```